docs/source/l1_to_l2b_e2e.rst

```diff
@@ -0,0 +1,198 @@
+
+.. code-block:: python
+
+    @pytest.mark.e2e
+    def test_l1_to_l2b(tvacdata_path, e2eoutput_path):
+        # Define input data paths
+        l1_datadir = os.path.join(tvacdata_path, "TV-36_Coronagraphic_Data", "L1")
+        l2b_datadir = os.path.join(tvacdata_path, "TV-36_Coronagraphic_Data", "L2b")
+        processed_cal_path = os.path.join(tvacdata_path, "TV-36_Coronagraphic_Data", "Cals")
+
+        # Create output directory if it does not exist
+        l2b_outputdir = os.path.join(e2eoutput_path, "l1_to_l2b_output")
+        if not os.path.exists(l2b_outputdir):
+            os.mkdir(l2b_outputdir)
+
+Calibration files such as non-linearity tables, dark current, flat fields, and bad pixel maps are loaded.
+
+.. code-block:: python
+
+        nonlin_path = os.path.join(processed_cal_path, "nonlin_table_240322.txt")
+        dark_path = os.path.join(processed_cal_path, "dark_current_20240322.fits")
+        flat_path = os.path.join(processed_cal_path, "flat.fits")
+        fpn_path = os.path.join(processed_cal_path, "fpn_20240322.fits")
+        cic_path = os.path.join(processed_cal_path, "cic_20240322.fits")
+        bp_path = os.path.join(processed_cal_path, "bad_pix.fits")
+
+The raw science data files and mock calibration files are defined.
+
+.. code-block:: python
+
+        l1_data_filelist = [os.path.join(l1_datadir, "{0}.fits".format(i)) for i in [90499, 90500]]
+        mock_cal_filelist = [os.path.join(l1_datadir, "{0}.fits".format(i)) for i in [90526, 90527]]
+        tvac_l2b_filelist = [os.path.join(l2b_datadir, "{0}.fits".format(i)) for i in [90529, 90531]]
+
+The calibration database (`caldb`) is initialized, and calibration entries are created.
+
+.. code-block:: python
+
+        pri_hdr, ext_hdr = mocks.create_default_headers()
+        ext_hdr["DRPCTIME"] = time.Time.now().isot
+        ext_hdr['DRPVERSN'] = corgidrp.__version__
+        mock_input_dataset = data.Dataset(mock_cal_filelist)
+
+        this_caldb = caldb.CalDB()  # Connection to the calibration database
+
+Non-linearity calibration, KGain, noise maps, flat field, and bad pixel maps are generated and stored in the calibration database.
+
+.. code-block:: python
+
+        # Nonlinearity calibration
+        nonlin_dat = np.genfromtxt(nonlin_path, delimiter=",")
+        nonlinear_cal = data.NonLinearityCalibration(nonlin_dat, pri_hdr=pri_hdr, ext_hdr=ext_hdr,
+                                                     input_dataset=mock_input_dataset)
+        nonlinear_cal.save(filedir=l2b_outputdir, filename="mock_nonlinearcal.fits")
+        this_caldb.create_entry(nonlinear_cal)
+
+        # KGain
+        kgain_val = 8.7
+        kgain = data.KGain(np.array([[kgain_val]]), pri_hdr=pri_hdr, ext_hdr=ext_hdr,
+                           input_dataset=mock_input_dataset)
+        kgain.save(filedir=l2b_outputdir, filename="mock_kgain.fits")
+        this_caldb.create_entry(kgain)
+
+The `walker.walk_corgidrp` function processes the L1 data.
```

Comment:
This is the most important line of code that does the actual processing. We should explain that this is the code that actually does the data processing. The rest is pipeline configuration.